pkg/optimizely/cache.go

```diff
@@ -95,7 +111,16 @@ func (c *OptlyCache) GetClient(sdkKey string) (*OptlyClient, error) {
 
 	// If we didn't "set" the key in this method execution then it was set in another thread.
 	// Recursively lookuping up the SDK key "should" only happen once.
-	return c.GetClient(sdkKey)
+	return c.GetClient(sdkKey, datafileAccessToken)
+}
+
+// UpdateConfigs is used to update config for all clients corresponding to a particular SDK key.
+func (c *OptlyCache) UpdateConfigs(sdkKey string) {
```

Comment:
For a particular webhook message, it is possible that there are multiple clients with same SDK key, but separate tokens.

Since the client key for private datafiles looks like `sdk_key:token`, we update all the configs corresponding to clients that begin with the given SDK key. Let me know if this is acceptable.